zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,870 @@
+                raise NotPipelineCompatible()
+            if isinstance(type_, Option):
+                type_ = type_.ty
+        except NotPipelineCompatible:
+            col = NonPipelineField(name, type_)
+        except TypeError:
+            col = NonNumpyField(name, type_)
+        else:
+            col = Column(type_.to_numpy_dtype().type)
+
+        columns[name] = col
+
+    name = expr._name
+    if expr._name is None:
+        name = next(_new_names)
+
+    # unicode is a name error in py3 but the branch is only hit
+    # when we are in python 2.
+    if PY2 and isinstance(name, unicode):  # noqa
+        name = name.encode('utf-8')
+
+    return type(name, (DataSet,), columns)
+
+
+def _check_resources(name, expr, resources):
+    """Validate that the exprssion and resources passed match up.
+
+    Parameters
+    ----------
+    name : str
+        The name of the argument we are checking.
+    expr : Expr
+        The potentially bound expr.
+    resources
+        The explicitly passed resources to compute expr.
+
+    Raises
+    ------
+    ValueError
+        If the resources to not match for an expression.
+    """
+    if expr is None:
+        return
+    bound = expr._resources()
+    if not bound and resources is None:
+        raise ValueError('no resources provided to compute %s' % name)
+    if bound and resources:
+        raise ValueError(
+            'explicit and implicit resources provided to compute %s' % name,
+        )
+
+
+def _check_datetime_field(name, measure):
+    """Check that a field is a datetime inside some measure.
+
+    Parameters
+    ----------
+    name : str
+        The name of the field to check.
+    measure : Record
+        The record to check the field of.
+
+    Raises
+    ------
+    TypeError
+        If the field is not a datetime inside ``measure``.
+    """
+    if not isinstance(measure[name], (Date, DateTime)):
+        raise TypeError(
+            "'{name}' field must be a '{dt}', not: '{dshape}'".format(
+                name=name,
+                dt=DateTime(),
+                dshape=measure[name],
+            ),
+        )
+
+
+class NoDeltasWarning(UserWarning):
+    """Warning used to signal that no deltas could be found and none
+    were provided.
+
+    Parameters
+    ----------
+    expr : Expr
+        The expression that was searched.
+    """
+    def __init__(self, expr):
+        self._expr = expr
+
+    def __str__(self):
+        return 'No deltas could be infered from expr: %s' % self._expr
+
+
+_valid_no_deltas_rules = 'warn', 'raise', 'ignore'
+
+
+def _get_deltas(expr, deltas, no_deltas_rule):
+    """Find the correct deltas for the expression.
+
+    Parameters
+    ----------
+    expr : Expr
+        The base expression.
+    deltas : Expr, 'auto', or None
+        The deltas argument. If this is 'auto', then the deltas table will
+        be searched for by walking up the expression tree. If this can not be
```

Comment:
"can not" -> "cannot" http://www.write.com/writing-guides/general-writing/word-choice/cannot-versus-can-not/